### PR TITLE
run functorch decomps after functionalization when enabled

### PR DESCRIPTION
**Pull request by bdhirsh**
This is a short-to-midterm fix for https://github.com/pytorch/pytorch/issues/83923. By running functionalization before decomps, we guarantee that functionalization won't have to see any primtorch view/inplace ops like `broadcast_in_dim`.

This will only really be a problem if there's a function in the decomposition table that decomposes a functional op into mutations. If that comes up later, we'll need to revisit https://github.com/pytorch/pytorch/issues/83923.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #83993
* __->__ #83992

### CODE REVIEW

File: aten/src/ATen/core/op_registration/op_registration_test.cpp [hunk 1]
```diff
@@ -2028,15 +2028,22 @@ TEST(NewOperatorRegistrationTest, testImplNoDefGetsCaught) {
...
+          ++dangling_prim_ops;
+      }
       error_str += "\t" + op_name.name;
       if (op_name.overload_name != "") {
           error_str += "." + op_name.overload_name;
       }
       error_str += "\n";
   }
-  ASSERT_EQ(danglingImpls.size(), 0) << error_str;
+  ASSERT_EQ(danglingImpls.size(), dangling_prim_ops) << error_str;
```
Comment (bdhirsh) [inline aten/src/ATen/core/op_registration/op_registration_test.cpp:2046]: sigh

**Review by ezyang — APPROVED** [review]
If we need to do more of these, let's expose enough API that these can be written in Python

**Comment by Chillee** [comment]
I'm kind of against doing this, tbh. I feel like this can be resolved by just doing decompositions after functionalization.

**Comment by bdhirsh** [comment]
Updated based on suggestions in that linked issue - now we'll just run decomps after functionalization.

I confirmed that with the repro command the errors also goes away.

**Pull request closed without merging**